components/net/http_cache.rs

```diff
@@ -428,12 +433,31 @@ fn handle_range_request(
                     },
                     _ => continue,
                 };
-                if res_beginning - 1 < beginning && res_end + 1 > end {
+                let (res_beginning_minus_one, res_end_plus_one) = {
+                    match (res_beginning.checked_sub(1), res_end.checked_add(1)) {
+                        (Some(minus_one), Some(plus_one)) => (minus_one, plus_one),
+                        _ => continue,
+                    }
+                };
+                if res_beginning_minus_one < beginning && res_end_plus_one > end {
                     let resource_body = &*partial_resource.body.lock().unwrap();
                     let requested = match resource_body {
                         &ResponseBody::Done(ref body) => {
-                            let b = beginning as usize - res_beginning as usize;
-                            let e = end as usize - res_beginning as usize + 1;
+                            let (b, e) = {
```

Comment:
Since we have already established that `res_beginning <= beginning` we do not need a checked_sub. Similarly, if we establish that `end >= res_beginning` then we don't need to check that operation either.